sdk/keyvault/Azure.Security.KeyVault.Keys/src/Page.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Text.Json;
+
+namespace Azure.Security.KeyVault.Keys
+{
+    public class Page<T> : Model
+        where T : Model
+    {
+        private T[] _items;
+        private Uri _nextLink;
+        private Func<T> _itemFactory;
+
+        internal Page(Func<T> itemFactory)
+        {
+            _itemFactory = itemFactory;
+        }
+
+        public ReadOnlySpan<T> Items { get => _items.AsSpan(); }
+
+        public Uri NextLink { get => _nextLink; }
+
+        internal override void ReadProperties(JsonElement json)
+        {
+            if (json.TryGetProperty("value", out JsonElement value))
+            {
+                _items = new T[value.GetArrayLength()];
+
+                int i = 0;
```

Comment:
nit:  Maybe we should consider a more meaningful name for this?  I realize that the scope is tiny and you can clearly see the usage below, but the single-letter-variable name makes me feel sad.  :)

Reply:
Making Jesse happy is important :) changing it to `index`